tutorials/nemo-retriever-synthetic-data-generation/README.md

```diff
@@ -0,0 +1,108 @@
+
+NeMo Retriever SDG can be run either from the command line, or using the [notebook example](notebooks/quickstart.ipynb) provided in this repository. Check the [Prerequisites](#prerequisites) section for instructions on generating an API key and installing libraries. To get started with the notebook, follow the [Notebook Quick Start](#run-pipeline-ipython-notebook) instructions. Otherwise, follow the [CLI Quick Start](#run-pipeline-cli) section.
+
+![NeMo Retriever SDG](figures/sdg_pipeline.png)
+
+#### Key Features
+
+* Quickly generate complex QA datasets from existing text documents for retriever model evaluation.
+* Output datasets can be formatted in [SQuAD (Stanford Question Answering Dataset)](https://huggingface.co/datasets/rajpurkar/squad) or [BEIR (Benchmarking Information Retrieval)](https://huggingface.co/datasets/BeIR/beir) format for easy integration into evaluation workflows.
+* Designed to integrate seamlessly with [NVIDIA NeMo Evaluator](https://developer.nvidia.com/nemo-microservices) microservice, currently in early access.
+
+
+## Quickstart
+
+### Prerequisites
+
+In order to use NeMo Retriever SDG, you will need access to NVIDIA’s API Catalog. Go to the [NGC Personal Key Manager](https://org.ngc.nvidia.com/setup) to generate a Personal Key that will allow you to access AI Foundation Models and Endpoints.
+
+To install the required libraries, navigate to the root directory of the project and run the following command in your notebook or command line:
+
+```
+$ pip install -r requirements.txt
+```
+
+Alternatively, you can use container `nvcr.io/nvidia/pytorch:24.01-py3`.
+
+```
+$ docker pull nvcr.io/nvidia/pytorch:24.01-py3
+
+$ docker run -it --rm --gpus all --ipc host --network host -v $(pwd):/workspace nvcr.io/nvidia/pytorch:24.01-py3
+
+/workspace# pip install -r requirements.txt
+/workspace# jupyter notebook
+```
+
+
+### Run Pipeline (iPython notebook)
+
+Navigate to the [quick start notebook](notebooks/quickstart.ipynb) and follow the instructions.
+
+### Run Pipeline (CLI)
+
+The pipeline can be run with datasets in either [SQuAD](https://huggingface.co/datasets/rajpurkar/squad) or rawdoc (only text and title) format. To test the pipeline, you can use the provided example data at ```sample_data_rawdoc.jsonl``` or ```sample_data_squad.json```.
+
+Navigate to the top level of this project directory and run the following command in your command line. It will take roughly 5-10 minutes.
+
+- `Rawdoc format`
+
+To use rawdoc format, provide your data in a `.jsonl` file. The structure of the data should follow this format: `{"text": <document>, "title": <title>}`.
+
+```
+PYTHONPATH=. python tutorials/synthetic-retrieval-evaluation-customization/main.py \
```

Comment:
Why is this `PYTHONPATH=.` prefix needed? Can you refactor to remove it?

Reply:
We dont need it, removed it.